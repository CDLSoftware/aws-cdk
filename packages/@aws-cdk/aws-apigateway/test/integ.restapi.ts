import cdk = require('@aws-cdk/cdk');
import apigateway = require('../lib');

class Test extends cdk.Stack {
    constructor(parent: cdk.App, id: string) {
        super(parent, id);

        const api = new apigateway.RestApi(this, 'my-api', {
            minimumCompressionSize: 0,
            autoDeployStageOptions: {
                stageName: 'test',
                description: 'testing stage'
            }
        });

        const v1 = api.newResource('api');

        const toys = v1.newResource('toys');
        toys.newMethod('GET');
        toys.newMethod('POST');
        toys.newMethod('PUT');

        const appliances = v1.newResource('appliances');
        appliances.newMethod('GET');

        const books = v1.newResource('books');
        books.newMethod('GET');
        books.newMethod('POST');
    }
}

const app = new cdk.App(process.argv);

new Test(app, 'test-apigateway-restapi');

process.stdout.write(app.run());