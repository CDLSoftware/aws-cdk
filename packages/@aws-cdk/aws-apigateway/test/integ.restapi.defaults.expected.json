{
  "Resources": {
    "myapi4C7BF186": {
      "Type": "AWS::ApiGateway::RestApi",
      "Properties": {
        "Name": "my-api"
      }
    },
    "myapiLatestDeployment24E142F763611a77ac8e9dca00e384f72144a2c9": {
      "Type": "AWS::ApiGateway::Deployment",
      "Properties": {
        "RestApiId": {
          "Ref": "myapi4C7BF186"
        }
      },
      "DependsOn": [
        "myapiGETF990CE3C"
      ],
      "DeletionPolicy": "Retain"
    },
    "myapiAutoDeployStageprod98CECBB1": {
      "Type": "AWS::ApiGateway::Stage",
      "Properties": {
        "RestApiId": {
          "Ref": "myapi4C7BF186"
        },
        "DeploymentId": {
          "Ref": "myapiLatestDeployment24E142F763611a77ac8e9dca00e384f72144a2c9"
        },
        "StageName": "prod"
      }
    },
    "myapiCloudWatchRole095452E5": {
      "Type": "AWS::IAM::Role",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com"
              }
            }
          ],
          "Version": "2012-10-17"
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn",
                ":",
                {
                  "Ref": "AWS::Partition"
                },
                ":",
                "iam",
                ":",
                "",
                ":",
                "aws",
                ":",
                "policy",
                "/",
                "service-role/AmazonAPIGatewayPushToCloudWatchLogs"
              ]
            ]
          }
        ]
      }
    },
    "myapiAccountEC421A0A": {
      "Type": "AWS::ApiGateway::Account",
      "Properties": {
        "CloudWatchRoleArn": {
          "Fn::GetAtt": [
            "myapiCloudWatchRole095452E5",
            "Arn"
          ]
        }
      },
      "DependsOn": [
        "myapi4C7BF186"
      ]
    },
    "myapiGETF990CE3C": {
      "Type": "AWS::ApiGateway::Method",
      "Properties": {
        "HttpMethod": "GET",
        "ResourceId": {
          "Fn::GetAtt": [
            "myapi4C7BF186",
            "RootResourceId"
          ]
        },
        "RestApiId": {
          "Ref": "myapi4C7BF186"
        },
        "AuthorizationType": "NONE",
        "Integration": {
          "Type": "MOCK"
        }
      }
    }
  },
  "Outputs": {
    "myapiEndpoint3628AFE3": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Ref": "myapi4C7BF186"
            },
            ".execute-api.",
            {
              "Ref": "AWS::Region"
            },
            ".amazonaws.com/",
            {
              "Ref": "myapiAutoDeployStageprod98CECBB1"
            },
            "/"
          ]
        ]
      },
      "Export": {
        "Name": "test-apigateway-restapi-defaults:myapiEndpoint3628AFE3"
      }
    }
  }
}